### PR TITLE
Feature/pt 2/themes

### PR DESCRIPTION
**Pull request by dungeon-slave**
end detailed page feature
use CachedImage instead default
3rd rework of firebase :(
rework MenuState
fix router
add theme feature
add dark & white themes
add & setup hive
add local theme saving
add app_button widget
core_ui fixes

https://github.com/dungeon-slave/food_app/assets/70040363/05220098-35a4-499d-b329-a209f705d571

### CODE REVIEW

File: core_ui/lib/src/widgets/app_button.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import 'package:core_ui/core_ui.dart';
+import 'package:flutter/material.dart';
+
+class AppButton extends StatelessWidget {
+  final String _text;
+  final void Function() _handler;
```
Comment (Alexey-Lebedzeu) [inline core_ui/lib/src/widgets/app_button.dart:6]: enter

File: core_ui/lib/src/widgets/app_error.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import 'package:flutter/widgets.dart';
+
+class AppError extends StatelessWidget {
+  final String _errorText;
+
+  const AppError({
+    required String errorText,
+    super.key,
+  }) : _errorText = errorText;
+  @override
```
Comment (Alexey-Lebedzeu) [inline core_ui/lib/src/widgets/app_error.dart:12]: enter

File: data/lib/di/data_di.dart [hunk 3]
```diff
@@ -19,10 +23,31 @@ class DataDI {
...
+    );
 
     appLocator.registerLazySingleton<FetchDishesUsecase>(
       () => FetchDishesUsecase(appLocator.get<DishesRepository>()),
     );
   }
+
+  void _initHive() {
+    appLocator.registerLazySingleton<HiveProvider>(
+      () => HiveProvider(),
```
Comment (Alexey-Lebedzeu) [inline data/lib/di/data_di.dart]: ,?

File: domain/lib/models/dish_model.dart [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
+  final List<String> ingredients;
 
   DishModel(
       {required this.imageRef,
       required this.name,
-      required this.type,
-      required this.price});
+      required this.price,
+      required this.description,
+      required this.ingredients});
```
Comment (Alexey-Lebedzeu) [inline domain/lib/models/dish_model.dart]: ,

File: domain/lib/usecase/theme/get_theme_usecase.dart [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import 'package:domain/repositories/theme_repository.dart';
+import 'package:domain/usecase/usecase.dart';
+
+class GetThemeUseCase implements UseCase<NoParams, bool> {
+  final ThemeRepository _themeRepository;
```
Comment (Alexey-Lebedzeu) [inline domain/lib/usecase/theme/get_theme_usecase.dart:5]: enter

File: domain/lib/usecase/theme/get_theme_usecase.dart [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+  const GetThemeUseCase({required ThemeRepository themeRepository})
+      : _themeRepository = themeRepository;
+  @override
```
Comment (Alexey-Lebedzeu) [inline domain/lib/usecase/theme/get_theme_usecase.dart:10]: enter

File: features/detailed_page/lib/src/ui/detailed_page_screen.dart [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+      ),
+      body: SingleChildScrollView(
+        padding: const EdgeInsets.only(bottom: AppDimens.padding100),
+        child: Column(
+          children: <Container>[
+            Container(
+              margin: const EdgeInsets.all(AppDimens.padding50),
+              decoration: BoxDecoration(
+                borderRadius: const BorderRadius.all(
+                    Radius.circular(AppDimens.padding10)),
```
Comment (Alexey-Lebedzeu) [inline features/detailed_page/lib/src/ui/detailed_page_screen.dart]: ,

File: features/detailed_page/lib/src/ui/detailed_page_screen.dart [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        decoration: BoxDecoration(
+          border: Border(
+            top: BorderSide(color: Theme.of(context).indicatorColor),
+          ),
+          boxShadow: <BoxShadow>[
+            BoxShadow(
+              blurStyle: BlurStyle.normal,
+              color: Theme.of(context).scaffoldBackgroundColor,
+            ),
+            // BoxShadow(
```
Comment (Alexey-Lebedzeu) [inline features/detailed_page/lib/src/ui/detailed_page_screen.dart]: ?

File: features/detailed_page/lib/src/ui/detailed_page_screen.dart [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+          children: <Widget>[
+            Text(
+              _model.price,
+              style: AppFonts.bold24.copyWith(
+                color: Theme.of(context).indicatorColor,
+              ),
+            ),
+            AppButton(
+              text: AppConstants.addToCart,
+              handler: () => {},
```
Comment (Alexey-Lebedzeu) [inline features/detailed_page/lib/src/ui/detailed_page_screen.dart]: ?

File: features/dishes_menu/lib/src/ui/dish_item.dart [hunk 10]
```diff
@@ -22,51 +18,59 @@ class DishItem extends StatefulWidget {
...
+                    color: Theme.of(context).indicatorColor,
                   ),
                 ),
-              ),
-            ],
-          ),
-        ],
+                AppButton(
+                  text: AppConstants.addToCart,
+                  handler: () => {},
```
Comment (Alexey-Lebedzeu) [inline features/dishes_menu/lib/src/ui/dish_item.dart]: ?

File: features/detailed_page/lib/src/ui/detailed_page_screen.dart (hunk 7 above)
Comment (dungeon-slave) [reply features/detailed_page/lib/src/ui/detailed_page_screen.dart]: + singlechild?

File: features/settings/lib/src/bloc/settings/settings_bloc_bloc.dart [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+part 'settings_bloc_event.dart';
+part 'settings_bloc_state.dart';
+
+class SettingsBloc extends Bloc<ThemeEvent, SettingsState> {
+  final SetThemeUseCase _setThemeUseCase;
+  final GetThemeUseCase _getThemeUseCase;
+
+  SettingsBloc(
+      {required SetThemeUseCase setThemeUseCase,
+      required GetThemeUseCase getThemeUseCase})
```
Comment (Alexey-Lebedzeu) [inline features/settings/lib/src/bloc/settings/settings_bloc_bloc.dart]: Поставь запятую следи внимательней  за форматированием

**Pull request merged**